src/DataStructures/DataBox/Prefixes.hpp

```diff
@@ -94,3 +100,56 @@ struct NormalDotNumericalFlux : db::DataBoxPrefix {
   static constexpr db::DataBoxString_t label = "NormalDotNumericalFlux";
 };
 }  // namespace Tags
+
+namespace Prefixes_detail {
+template <typename, template <typename...> class Prefix, typename Tag,
+          typename... Args>
+struct add_tag_prefix_impl;
+
+template <template <typename...> class Prefix, typename Tag, typename... Args>
+struct add_tag_prefix_impl<std::false_type, Prefix, Tag, Args...> {
+  using type = Prefix<Tag, Args...>;
+};
+
+template <template <typename...> class Prefix, typename Tag, typename... Args>
+struct add_tag_prefix_impl<std::true_type, Prefix, Tag, Args...> {
+  using type = Prefix<
+      Tags::Variables<tmpl::transform<typename Tag::tags_list,
+                                      tmpl::bind<Prefix, tmpl::_1, Args...>>>,
+      Args...>;
+};
+
+template <typename>
+struct remove_tag_prefix_impl;
+
+template <typename UnprefixedTag, template <typename...> class Prefix,
+          typename... Args>
+struct remove_tag_prefix_impl<Prefix<UnprefixedTag, Args...>> {
+  static_assert(cpp17::is_base_of_v<db::DataBoxTag, UnprefixedTag>,
+                "Unwrapped tag is not a DataBoxTag");
+  using type = UnprefixedTag;
+};
+
+template <typename... VariablesTags, template <typename...> class Prefix,
+          typename... Args>
+struct remove_tag_prefix_impl<
+    Prefix<Tags::Variables<tmpl::list<VariablesTags...>>, Args...>> {
+  using type =
+      Tags::Variables<tmpl::transform<tmpl::list<VariablesTags...>,
+                                      remove_tag_prefix_impl<tmpl::_1>>>;
+};
+}  // namespace Prefixes_detail
+
+/// \ingroup DataBoxTags
+/// Wrap `Tag` in `Prefix<_, Args...>`, also wrapping variables tags
+/// if `Tag` is a `Tags::Variables`.
+template <template <typename...> class Prefix, typename Tag, typename... Args>
```

Comment:
These are a really nice complement to the `wrap_tags_in`, which is currently in `DataBoxTag`. I currently think that's the better place for these since we might end up with prefix tags floating around other places and `DataBox` is required for any work with Tags. That's my current reasoning anyway :)